content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1422 @@
+use other [payment types](payment-types.html) like [Escrow](https://xrpl.org/escrow.html) or [Payment Channels](https://xrpl.org/payment-channels.html). However, it provides a foundation 
+that you can implement those and other features on top of.
+
+In addition to the above features, you'll also learn a bit about Events, IPC (inter-process-communication) 
+and asynchronous (async) code in JavaScript.
+
+## Steps
+
+### 0. Project setup - Hello World
+
+1. To initialize the project, create a package.json file with the following content:
+
+```json
+{
+  "name": "xrpl-javascript-desktop-wallet",
+  "version": "1.0.0",
+  "license": "MIT",
+  "scripts": {
+    "start": "electron ./index.js"
+  },
+  "dependencies": {
+    "async": "^3.2.4",
+    "fernet": "^0.4.0",
+    "node-fetch": "^2.6.9",
+    "pbkdf2-hmac": "^1.1.0",
+    "open": "^8.4.0",
+    "toml": "^3.0.0",
+    "xrpl": "^2.6.0"
+  },
+  "devDependencies": {
+    "electron": "22.3.2"
+  }
+}
+```
+
+Here we define the libraries our application will use in the `dependencies` section as well as shortcuts for running our application in the `scripts` section. 
+
+2. After you create your package.json file, install those dependencies by running the following command: 
+```console
+npm install
+```
+
+This installs the Electron Framework, the xrpl.js client library and a couple of helpers we are going to need for our
+application to work. 
+
+3. In the root folder, create an `index.js` file with the following content:
+
+```javascript
+const { app, BrowserWindow } = require('electron')
+
+const path = require('path')
+
+/**
+ * This is our main function, it creates our application window, preloads the code we will need to communicate
+ * between the renderer Process and the main Process, loads a layout and performs the main logic
+ */
+const createWindow = () => {
+
+  // Creates the application window
+  const appWindow = new BrowserWindow({
+    width: 1024,
+    height: 768
+  })
+
+  // Loads a layout
+  appWindow.loadFile(path.join(__dirname, 'view', 'template.html'))
+
+  return appWindow
+}
+
+// Here we have to wait for the application to signal that it is ready
+// to execute our code. In this case we just create a main window.
+app.whenReady().then(() => {
+  createWindow()
+})
+```
+
+4. Make a new folder named `view` in the root directory of the project.
+
+5. Now, inside the `view` folder, add a `template.html` file with the following content:
+
+```html
+<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset="UTF-8" />
+  <meta http-equiv="Content-Security-Policy" content="default-src 'self'; script-src 'self'" />
+  <meta http-equiv="X-Content-Security-Policy" content="default-src 'self'; script-src 'self'" />
+  <title>XRPL Wallet Tutorial (JavaScript / Electron)</title>
+</head>
+
+<body>
+
+  <h3>Build a XRPL Wallet</h3>
+  <span>Hello world!</span>
+
+</body>
+
+</html>
+```
+6. Now, start the application with the following command:
+
+```console
+npm run start
+```
```

Comment:
Adding in expected values so they know they did it right :)

```suggestion
```

You should see a window appear that displays the text "Build a XRPL Wallet" and "Hello world!"
```